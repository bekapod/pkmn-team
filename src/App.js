// @flow
import React from "react";
import { Switch, Route } from "react-router-dom";
import DashboardContainer from "./containers/Dashboard";
import TeamBuilderContainer from "./containers/TeamBuilder";

const App = () => (
  <Switch>
    <Route exact path="/" component={DashboardContainer} />
    <Route path="/team/create/" component={TeamBuilderContainer} />
  </Switch>
);

export default App;
