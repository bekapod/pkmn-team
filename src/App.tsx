import React from "react";
import { Route, Switch } from "react-router-dom";
import DashboardContainer from "./containers/Dashboard";
import TeamBuilderContainer from "./containers/TeamBuilder";

const App = () => (
  <Switch>
    <Route exact={true} path="/" component={DashboardContainer} />
    <Route path="/team/create/" component={TeamBuilderContainer} />
    <Route path="/team/edit/:teamId" component={TeamBuilderContainer} />
  </Switch>
);

export default App;
