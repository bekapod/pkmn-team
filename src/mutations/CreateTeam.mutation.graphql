mutation CreateTeam($name: String!) {
  createTeam(object: { name: $name }) {
    ...TeamFragment
  }
}
