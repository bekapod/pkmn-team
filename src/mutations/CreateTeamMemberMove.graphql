mutation CreateTeamMemberMove($memberId: uuid!, $moveId: uuid!, $order: Int!) {
  createTeamMemberMove(
    object: { team_member_id: $memberId, move_id: $moveId, order: $order }
  ) {
    ...TeamMemberMoveFragment
    team_member {
      id
      team {
        id
      }
    }
  }
}
