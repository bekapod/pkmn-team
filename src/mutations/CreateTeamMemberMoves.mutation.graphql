mutation CreateTeamMemberMoves($moves: [team_member_move_insert_input!]!) {
  createTeamMemberMoves(
    objects: $moves
    on_conflict: {
      constraint: team_member_move_team_member_id_move_id_key
      update_columns: [order]
    }
  ) {
    returning {
      ...TeamMemberMoveFragment
      team_member {
        id
        team {
          id
        }
      }
    }
  }
}
