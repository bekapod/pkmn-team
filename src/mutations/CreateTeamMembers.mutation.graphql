mutation CreateTeamMembers($members: [team_member_insert_input!]!) {
  createTeamMembers(objects: $members) {
    returning {
      ...TeamMemberFragment
    }
  }
}
