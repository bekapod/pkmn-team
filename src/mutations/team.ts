import gql from "graphql-tag";

export const createTeam = gql`
  mutation($name: String!, $pokedexIds: [Int!]!) {
    createTeam(name: $name, pokedexIds: $pokedexIds) {
      id
    }
  }
`;
