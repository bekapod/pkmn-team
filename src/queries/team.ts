import gql from "graphql-tag";

export const getAllTeams = gql`
  query {
    allTeams {
      id
      name
      createdAt
      members {
        id
        pokemon {
          pokedexId
          name
          types
          sprite
        }
      }
    }
  }
`;
