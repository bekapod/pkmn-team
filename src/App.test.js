// @flow
import React from "react";
import MemoryRouter from "react-router-dom/MemoryRouter";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { MockedProvider } from "react-apollo/test-utils";
import wait from "waait";
import configureStore from "redux-mock-store";
import App from "./App";
import { getAllTeams } from "./queries/team";
import { getAllPokemon } from "./queries/pokemon";
import { createTeam } from "./mutations/team";

const mocks = [
  {
    request: {
      query: getAllTeams
    },
    result: {
      loading: false,
      data: {
        allTeams: [
          {
            id: "cji6gz8gwhblk0a9639btq2hd",
            name: "Starters Team",
            createdAt: "2018-06-08T21:15:14.723Z",
            members: [
              {
                id: "cji6gz8gwhbll0a96aahx3ivv",
                pokemon: {
                  pokedexId: 1,
                  name: "bulbasaur",
                  types: ["POISON", "GRASS"],
                  sprite: "1.png"
                }
              },
              {
                id: "cji6gz8gwhblm0a96eja18t10",
                pokemon: {
                  pokedexId: 4,
                  name: "charmander",
                  types: ["FIRE"],
                  sprite: "4.png"
                }
              },
              {
                id: "cji6gz8gwhbln0a96q7wmx9zj",
                pokemon: {
                  pokedexId: 7,
                  name: "squirtle",
                  types: ["WATER"],
                  sprite: "7.png"
                }
              },
              {
                id: "cji6gz8gwhblo0a96wgoki379",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              }
            ]
          },
          {
            id: "cji6kxx7gp5n80a96mgw7gc45",
            name: "Pikachu Team",
            createdAt: "2018-06-08T23:06:11.936Z",
            members: [
              {
                id: "cji6kxx7hp5n90a96r5rpx50u",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kxx7hp5na0a96hb6e0643",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kxx7hp5nb0a9629wdve5s",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kxx7hp5nc0a969th9ll1f",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kxx7hp5nd0a968aj6f6qj",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kxx7hp5ne0a964tf72wol",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              }
            ]
          },
          {
            id: "cji6kzy8pprbq0a964rc3ikk3",
            name: "Pikachu Team",
            createdAt: "2018-06-08T23:07:46.587Z",
            members: [
              {
                id: "cji6kzy8pprbr0a967jhcik0n",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kzy8pprbs0a96349zeq46",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              },
              {
                id: "cji6kzy8pprbt0a96v5isjwvt",
                pokemon: {
                  pokedexId: 25,
                  name: "pikachu",
                  types: ["ELECTRIC"],
                  sprite: "25.png"
                }
              }
            ]
          }
        ]
      }
    }
  },

  {
    request: {
      query: getAllPokemon
    },
    result: {
      loading: false,
      data: {
        allPokemon: [
          {
            pokedexId: 25,
            name: "pikachu",
            types: ["ELECTRIC"],
            sprite: "25.png"
          }
        ]
      }
    }
  },

  {
    request: {
      mutation: createTeam,
      variables: {
        name: "Test Team",
        pokedexIds: [25]
      }
    },
    result: {
      createTeam: {
        id: "1"
      }
    }
  }
];

describe("<App />", () => {
  const mockStore = configureStore();

  it("renders dashboard", async () => {
    const tree = renderer.create(
      <Provider store={mockStore({})}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <MemoryRouter initialEntries={["/"]}>
            <App />
          </MemoryRouter>
        </MockedProvider>
      </Provider>
    );

    await wait(0);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders create a team form", () => {
    const tree = renderer.create(
      <Provider store={mockStore({})}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <MemoryRouter initialEntries={["/team/create/"]}>
            <App />
          </MemoryRouter>
        </MockedProvider>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
