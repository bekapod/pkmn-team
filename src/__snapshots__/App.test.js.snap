// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders create a team form 1`] = `
Array [
  <h1
    className="Heading-sc-1a24dwr-0 eiJqiJ"
  >
    Create a Team
  </h1>,
  <div
    className="SectionContainer-sc-1h7hs07-0 lhxZhT"
  >
    <div
      className="CenteredRow-sc-1446zpg-0 bqMlyu"
    >
      <input
        aria-label="Choose a team name"
        className="TextInput-sc-17mm9lu-0 GiantInput-sc-1xpf4pb-0 dwkaeV"
        onChange={[Function]}
        placeholder="Choose a team name"
      />
    </div>
    <div
      className="CenteredRow-sc-1446zpg-0 dFcFNQ"
    >
      <div>
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="downshift-0-label"
          aria-owns={null}
          className="Autocomplete-sc-18qws42-1 jEZDkd"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            autoComplete="off"
            className="TextInput-sc-17mm9lu-0 GiantInput-sc-1xpf4pb-0 dwkaeV"
            id="downshift-0-input"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Choose a Pokemon"
            value=""
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`<App /> renders dashboard 1`] = `
Array [
  <h1
    className="Heading-sc-1a24dwr-0 eiJqiJ"
  >
    My Teams
  </h1>,
  <div
    className="SectionContainer-sc-1h7hs07-0 lhxZhT"
  >
    <div
      className="CenteredRow-sc-1446zpg-0 dFcFNQ"
    >
      <a
        className="Cta__CtaInternalLink-sc-1qsmnol-0 isVVbW"
        href="/team/create/"
        onClick={[Function]}
      >
        Create a team
      </a>
    </div>
    <div
      className="TeamGrid__CardGrid-sc-1wwzl6c-0 dDkaIx"
    >
      <article
        className="Card__CardWrapper-sc-1prdgex-0 hFLESf"
      >
        <header
          className="Card__CardHeader-sc-1prdgex-1 hpFuRB"
        >
          <h2
            className="CardHeading-esgf7h-0 hqHCZd"
          >
            Starters Team
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-1prdgex-2 cyIVmL"
        >
          <div
            className="CardMeta__CardMetaWrapper-sc-6wqr52-0 hgGiwQ"
          >
            <div
              className="CardMeta__CardMetaItem-sc-6wqr52-1 bDCDfU"
            >
              <span
                className="CardMeta__CardMetaLabel-sc-6wqr52-2 bZDTki"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-sc-6wqr52-3 dquxYs"
              >
                4
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-sc-6wqr52-1 bDCDfU"
            >
              <span
                className="CardMeta__CardMetaLabel-sc-6wqr52-2 bZDTki"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-sc-6wqr52-3 dquxYs"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 kgmVFY"
          >
            <img
              alt="bulbasaur sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/1.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #1 Bulbasaur
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 dCWWOO"
                    type="GRASS"
                  >
                    GRASS
                  </span>
                </li>
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 daqRLF"
                    type="POISON"
                  >
                    POISON
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 eLaVlL"
          >
            <img
              alt="charmander sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/4.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #4 Charmander
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 lnMICI"
                    type="FIRE"
                  >
                    FIRE
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 iUdtBD"
          >
            <img
              alt="squirtle sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/7.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #7 Squirtle
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 HMLnI"
                    type="WATER"
                  >
                    WATER
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
      <article
        className="Card__CardWrapper-sc-1prdgex-0 hFLESf"
      >
        <header
          className="Card__CardHeader-sc-1prdgex-1 iQgYsh"
        >
          <h2
            className="CardHeading-esgf7h-0 hqHCZd"
          >
            Pikachu Team
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-1prdgex-2 cyIVmL"
        >
          <div
            className="CardMeta__CardMetaWrapper-sc-6wqr52-0 hgGiwQ"
          >
            <div
              className="CardMeta__CardMetaItem-sc-6wqr52-1 bDCDfU"
            >
              <span
                className="CardMeta__CardMetaLabel-sc-6wqr52-2 bZDTki"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-sc-6wqr52-3 dquxYs"
              >
                6
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-sc-6wqr52-1 bDCDfU"
            >
              <span
                className="CardMeta__CardMetaLabel-sc-6wqr52-2 bZDTki"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-sc-6wqr52-3 dquxYs"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
      <article
        className="Card__CardWrapper-sc-1prdgex-0 hFLESf"
      >
        <header
          className="Card__CardHeader-sc-1prdgex-1 ixIMHb"
        >
          <h2
            className="CardHeading-esgf7h-0 hqHCZd"
          >
            Pikachu Team
          </h2>
        </header>
        <div
          className="Card__CardContent-sc-1prdgex-2 cyIVmL"
        >
          <div
            className="CardMeta__CardMetaWrapper-sc-6wqr52-0 hgGiwQ"
          >
            <div
              className="CardMeta__CardMetaItem-sc-6wqr52-1 bDCDfU"
            >
              <span
                className="CardMeta__CardMetaLabel-sc-6wqr52-2 bZDTki"
              >
                Pkmn
              </span>
              <span
                className="CardMeta__CardMetaValue-sc-6wqr52-3 dquxYs"
              >
                3
              </span>
            </div>
            <div
              className="CardMeta__CardMetaItem-sc-6wqr52-1 bDCDfU"
            >
              <span
                className="CardMeta__CardMetaLabel-sc-6wqr52-2 bZDTki"
              >
                Created
              </span>
              <span
                className="CardMeta__CardMetaValue-sc-6wqr52-3 dquxYs"
              >
                8/6/18
              </span>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
          <div
            className="PokemonLine__Row-z6g3b0-0 geQAmh"
          >
            <img
              alt="pikachu sprite"
              className="PokemonLine__RowImage-z6g3b0-1 ewsyct"
              src="/sprites/25.png"
            />
            <div>
              <div
                className="PokemonLine__RowTitle-z6g3b0-2 huXkVG"
              >
                #25 Pikachu
              </div>
              <ul
                className="InlineList-bwsyp4-0 bKIwJu"
              >
                <li>
                  <span
                    className="TypeTag-sc-19epxam-0 eoDTXf"
                    type="ELECTRIC"
                  >
                    ELECTRIC
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>,
]
`;
